cgaal-cli/src/args.rs

```diff
@@ -1,86 +1,243 @@
-use clap::{App, Arg};
+use crate::options::{
+    CliOptions, FormulaFormat, ModelFormat, SearchStrategyOption, SubcommandOption,
+};
+use clap::{App, AppSettings, Arg, ArgMatches, SubCommand};
+use git_version::git_version;
+
+const PKG_NAME: &str = env!("CARGO_PKG_NAME");
+const AUTHORS: &str = env!("CARGO_PKG_AUTHORS");
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+const GIT_VERSION: &str = git_version!(fallback = "unknown");
+
+/// Define and parse command line arguments
+pub fn parse_arguments() -> Result<CliOptions, String> {
+    let version_text = format!("{} ({})", VERSION, GIT_VERSION);
+    let string = AUTHORS.replace(':', "\n");
+    let mut app = App::new(PKG_NAME)
+        .version(version_text.as_str())
+        .author(string.as_str())
+        .arg(
+            Arg::with_name("log_filter")
+                .short("l")
+                .long("log-filter")
+                .env("RUST_LOG")
+                .default_value("warn")
+                .global(true)
+                .help("Comma separated list of filter directives"),
+        )
+        .subcommand(
+            SubCommand::with_name("check")
+                .about("Checks satisfiability of an ATL property on a CGS")
+                .add_positional_model_path_arg()
+                .add_model_format_arg()
+                .add_positional_formula_arg()
```

Comment:
From claps POV this is just two strings. If we allow swapping them, then we need some code to determine which is which before usage, which may include attempting to parse the files since the extension `.json` could be both model and formula and we also allow the files to have no file extension (in which case `--model-format` has to be specified). It's much easier to not allow it.

Reply:
We also want to be future-proof. For #217, we might want to allow multiple formulas in multiple files. In that case, it will also be much easier if the first path is the model and the remaining following paths are formulas.

Reply:
I forgot about JSON, we'll push request to issue #225